pyroma/tests.py

```diff
@@ -245,47 +203,49 @@ def test_markdown(self):
 
 
 class PyPITest(unittest.TestCase):
-    def test_distribute(self):
-        real_urlopen = urllib.urlopen
-        real_server_proxy = xmlrpclib.ServerProxy
-        try:
-            xmlrpclib.ServerProxy = ProxyStub(
-                "distributedata.py", xmlrpclib.ServerProxy, False
-            )
-            urllib.urlopen = urlopenstub
-            data = pypidata.get_data("distribute")
-            rating = rate(data)
-
-            self.assertEqual(
-                rating,
-                (
-                    9,
-                    [
-                        "The classifiers should specify what minor versions of Python "
-                        "you support as well as what major version.",
-                        "You should have three or more owners of the project on PyPI.",
-                    ],
-                ),
-            )
-        finally:
-            xmlrpclib.ServerProxy = real_server_proxy
-            urllib.urlopen = real_urlopen
 
-    def test_complete(self):
-        real_urlopen = urllib.urlopen
-        real_server_proxy = xmlrpclib.ServerProxy
-        try:
-            xmlrpclib.ServerProxy = ProxyStub(
-                "completedata.py", xmlrpclib.ServerProxy, False
-            )
-            urllib.urlopen = urlopenstub
-            data = pypidata.get_data("complete")
-            rating = rate(data)
-
-            self.assertEqual(rating, (10, []))
-        finally:
-            xmlrpclib.ServerProxy = real_server_proxy
-            urllib.urlopen = real_urlopen
+    @mock.patch('xmlrpc.client.ServerProxy', proxystub)
+    @mock.patch('pyroma.pypidata._get_project_data')
+    def test_distribute(self, projectdatamock):
+        datafile = resource_filename(
+            __name__, os.path.join("testdata", "jsondata", "distribute.json")
+        )
+        with open(datafile, 'rt') as file:
```

Comment:
```suggestion
        with open(datafile, 'rt', encoding="UTF-8") as file:
```

As described in #28 and fixed in PR #53 , otherwise, the tests will fail on Windows if this file has any non-ASCII characters, or the current locale's codepage is not ASCII-compatible.

Reply:
Since that file doesn't have non-ascii characters it's really not necessary.